node/src/components/block_accumulator.rs

```diff
@@ -426,7 +380,11 @@ impl BlockAccumulator {
             None => return Effects::new(),
         };
 
-        match acceptor.register_finality_signature(finality_signature, sender) {
+        match acceptor.register_finality_signature(
+            finality_signature,
+            sender,
+            self.chainspec.core_config.validator_slots,
```

Comment:
`validator_slots` is the only parameter from the chainspec that we need in `BlockAccumulator`, so maybe consider passing this particular parameter instead of the entire chainspec [here](https://github.com/casper-network/casper-node/pull/3709/files#diff-ea02e6cf88dd93f39b659e8772c87219d3f5ebb962a0c6833cbb1baac90d80e6R333).

This will also slightly simplify the tests since you can get rid of the [chainspec_mock()](https://github.com/casper-network/casper-node/pull/3709/files#diff-0e1032d5ddaa0f1190cfdfae21e9dfe5c03b7cb33b32711d31b5be67e766c785R1750-R1752) function.

Reply:
Done